ipatests/test_integration/create_caless_pki.py

```diff
@@ -204,6 +204,21 @@ def profile_server(builder, ca_nick, ca,
             critical=False
         )
 
+    if wildcard:
+        server_domain = server1.split('.', 1)[1]
+        if domain != server_domain:
+            builder = builder.add_extension(
+                x509.SubjectAlternativeName([
+                     x509.DNSName(u'*.' + domain),
+                     x509.DNSName(u'*.' + server_domain),
+                ]),
+                critical=False,
+            )
+        else:
+            builder = builder.add_extension(
+                x509.SubjectAlternativeName([x509.DNSName(u'*.' + domain)]),
+                critical=False,
+            )
```

Comment:
I think the logic is fine, but a few comments:

1. partially build the GeneralName list, then append to it if the condition matches (eliminate duplicate code)
2. if for some reason there is no `.` in `domain`, we don't want to crash when indexing the split result at `[1]`.

So, something like:

```python
if wildcard:
  names = [x509.DNSName(u'*.' + domain)]
  server_split = server1.split('.', 1)
  if len(server_split) == 2 and domain != server_split[1]
    names.append(x509.DNSName(u'*.' + server_split[1])
  builder = builder.add_extension(x509.SubjectAlternativeName(names), critical=False)
```